app/components/shared/ShowMoreFooter.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Relay from 'react-relay/classic';
+import classNames from 'classnames';
+
+import Button from './Button';
+import Spinner from './Spinner';
+
+class ShowMoreFooter extends React.PureComponent {
+  static propTypes = {
+    connection: PropTypes.shape({
+      pageInfo: PropTypes.shape({
+        hasNextPage: PropTypes.bool.isRequired
+      }).isRequired
+    }),
+    onShowMore: PropTypes.func.isRequired,
+    label: PropTypes.string,
+    loading: PropTypes.bool,
+    searching: PropTypes.bool,
+    className: PropTypes.string
+  };
+
+  static defaultProps = {
+    loading: false,
+    searching: false,
+    className: 'px3 py2'
+  };
+
+  render() {
+    const { connection, loading, searching } = this.props;
+
+    // don't show any footer if we're searching
+    if (searching) {
+      return null;
+    }
+
+    // don't show any footer if we haven't ever loaded
+    // any items, or if there's no next page
+    if (!connection || !connection.pageInfo.hasNextPage) {
+      return null;
+    }
+
+    let footerContent = (
+      <Button
+        outline={true}
+        theme="default"
+        onClick={this.props.onShowMore}
+      >
+        Show more{this.props.label ? ` ${this.props.label}` : ''}…
+      </Button>
+    );
+
+    // show a spinner if we're loading more
+    if (loading) {
+      footerContent = <Spinner style={{ margin: 9.5 }} />;
+    }
```

Comment:
I don’t really like Ruby’s implicit return values 🙅🏻 but I do kind of wish there was *a* way to return something from a conditional. (I think this is why I like ternary statements, to some peoples’ chagrin)

I don’t believe this approach has any major cost (all this will do is call “React.createElement”, which I don’t believe will end up doing anything meaningful until the element is actually mounted, which it thus won’t be.